gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -182,12 +255,32 @@ def _merge_headers(known_args, pipeline_args, pipeline_mode):
         headers,
         known_args.split_alternate_allele_info_fields,
         known_args.allow_incompatible_records)
+    if original_input_pattern:
```

Comment:
I found this check confusing. 
By just looking at this, I have no idea why headers are not merged if original_input_pattern is None, nor why original_input_pattern can be even None. I have to go and read the piece of the code that sets `original_input_pattern` in order to understand it. Could you use a different conditional variable, something explaining the situation better, perhaps something indicating annotation step is undergoing.